src/game/ButtonsUsed.js

```diff
@@ -0,0 +1,14 @@
+/**
+ Data file used to send the amount of buttons used and their name to the controller.
+ This file needs to be updated for each game-mode.
+*/
+
+const buttonsUsed = {
+  knockOff: ['Super Heavy'],
+  knockOffDynamic: ['Super Heavy'],
+  knockOffRandom: ['Super Heavy'],
+  knockOffWander: ['Super Heavy'],
+  testGamemode: ['Test Ability 1', 'Test Ability 2'],
+};
+
+export default buttonsUsed;
```

Comment:
This couldn't be inside Gameode since GamemodeHandler needs to read which buttons a gamemode is using and then create an instance with this data. This is done before the game is started and the gamemode  class initialized, before initialization we can't access any of its fields and trying to initialize the class earlier causes crashes. This was the easiest fix.

Reply:
Then I would suggest you implement this like the gamemodeResources within the GamemodeHandler. Otherwise it can become annoying to browse multiple files to create one gamemode.

Reply:
Yea  good idea, I wasn't sure where to put the button data